### PR TITLE
cleanup secrets holding cloud-init data

### PR DESCRIPTION
**Pull request by paha**
- reorginize cleanup functions
- delete cloud-init-* secrets older then 5 hours

### CODE REVIEW

File: testgrid/tgrun/pkg/runner/runner.go [hunk 1]
```diff
@@ -296,7 +296,7 @@ echo "running sonobuoy"
 curl -L --output ./sonobuoy.tar.gz https://github.com/vmware-tanzu/sonobuoy/releases/download/v0.18.3/sonobuoy_0.18.3_linux_amd64.tar.gz
 tar xzvf ./sonobuoy.tar.gz
 
-if [ -z $DOCKERHUB_USER ]; then
+if [ -n $DOCKERHUB_PASS ]; then
```
Comment (emosbaugh) [inline testgrid/tgrun/pkg/runner/runner.go:299]: do you need to quote this?

File: testgrid/tgrun/pkg/runner/cleanup.go [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+
+// CleanUpSecrets removes stale cloud-init configurations stored in secrets and prevents errors on interupted runs
+func cleanupSecrets(clientset *kubernetes.Clientset) error {
+	secrets, err := clientset.CoreV1().Secrets(Namespace).List(metav1.ListOptions{})
+	if err != nil {
+		return errors.Wrap(err, "failed to get secrets list")
+	}
+
+	for _, sec := range secrets.Items {
+		// Delete stale secrets older then 5 hours
```
Comment (emosbaugh) [inline testgrid/tgrun/pkg/runner/cleanup.go:129]: seems like an hour should be ok

File: testgrid/tgrun/pkg/runner/cleanup.go [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+		if len(sec.Name) > 6 && sec.Name[:5] == "cloud" && time.Since(sec.CreationTimestamp.Time).Hours() > 5 {
+			clientset.CoreV1().Secrets(Namespace).Delete(sec.Name, &metav1.DeleteOptions{})
```
Comment (emosbaugh) [inline testgrid/tgrun/pkg/runner/cleanup.go:131]: you should probably handle this error. at least output a message

File: testgrid/tgrun/pkg/runner/cleanup.go [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+			// However local directory is left with tmpimage file, removing
+			// 523M    /var/openebs/local/pvc-xxx
+			err := os.RemoveAll(localPath)
+			if err != nil {
+				fmt.Printf("Failed to delete %s; ERROR: %s", localPath, err)
+			}
+		} else if pv.ObjectMeta.DeletionTimestamp != nil {
+			// cleaning pv stack in Terminating state
+			pv.ObjectMeta.SetFinalizers(nil)
+			p, _ := clientset.CoreV1().PersistentVolumes().Update(&pv)
```
Comment (emosbaugh) [inline testgrid/tgrun/pkg/runner/cleanup.go:114]: you should probably handle this error. at least output a message

File: testgrid/tgrun/pkg/runner/cleanup.go [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+// CleanUpSecrets removes stale cloud-init configurations stored in secrets and prevents errors on interupted runs
+func cleanupSecrets(clientset *kubernetes.Clientset) error {
+	secrets, err := clientset.CoreV1().Secrets(Namespace).List(metav1.ListOptions{})
+	if err != nil {
+		return errors.Wrap(err, "failed to get secrets list")
+	}
+
+	for _, sec := range secrets.Items {
+		// Delete stale secrets older then 5 hours
+		if len(sec.Name) > 6 && sec.Name[:5] == "cloud" && time.Since(sec.CreationTimestamp.Time).Hours() > 5 {
```
Comment (emosbaugh) [inline testgrid/tgrun/pkg/runner/cleanup.go:130]: this is fine but maybe we should just add a label or something

**Review by emosbaugh — APPROVED** [review]

File: testgrid/tgrun/pkg/runner/cleanup.go (hunk 2 above)
Comment (paha) [reply testgrid/tgrun/pkg/runner/cleanup.go:129]: The intent is to leave it for a bit, if there is a need to examine what the resulting cloud-init content was, for potential debugging.

**Pull request closed without merging**